src/components/App/PuzzleGenerate/index.tsx

```diff
@@ -0,0 +1,69 @@
+import * as React from 'react';
+import { useEffect, useState } from 'react';
+import styles from './PuzzleGenerate.module.scss';
+import { Button } from '../../Global';
+import { getAllPuzzles } from '../../../services';
+import {createPuzzleInstance} from "../../../services/puzzleInstance";
+
+const PuzzleGenerate = () => {
+  const handleSubmit = event => {
+    event.preventDefault();
+    const hint = event.target[0].value;
+    const latitude = event.target[1].value;
+    const longitude = event.target[2].value;
+    const address = event.target[3].value;
+    const puzzle = event.target[4].value;
+
+    console.log(hint, latitude, longitude, address, puzzle);
+    createPuzzleInstance(puzzle, longitude, latitude, address, hint).then(r => console.log(r));
+  };
+
+  const [puzzleList, setPuzzleList] = useState([]);
+
+  useEffect(() => {
+    const fetchPuzzles = async () => {
+      setPuzzleList(await getAllPuzzles());
+    };
+    fetchPuzzles();
+  }, []);
+
+  return (
+    <>
+      <form className={styles.form} onSubmit={event => handleSubmit(event)}>
```

Comment:
I prefer the pattern with state variables and then a function that's called on submit. Doing it this way and then using arrays feels iffy

Reply:
We should use the Input component when we switch it over, helps with consistent styling, too